### PR TITLE
Add docs for Genie integration

### PR DESCRIPTION
**Pull request by qti-kromero**
### Description
Adds documentation of Genie integration, including information on options available and how to use the integration

### Motivation and Context
Awareness of feature and how people can use it

### CODE REVIEW

File: docs/execution_providers/QNN-ExecutionProvider.md [hunk 1]
```diff
@@ -580,6 +593,81 @@ del session
 ort.unregister_execution_provider_library(ep_registration_name)
 ```
 
+## Running an LLM model with QNN EP's Genie backend
+
+The QNN EP includes a Genie execution pathway for accelerated LLM inference on Windows ARM64 devices.
+Genie is Qualcomm's generative AI framework (part of the QAIRT SDK) that the QNN EP can delegate token-to-logit inference
+for auto-regressive LLM generation to using pre-compiled DLC (Deep Learning Container) models.
+
+For dialog functionalities, onnxruntime-qnn should be used with onnxruntime-genai.
```
Comment (qti-kromero) [inline docs/execution_providers/QNN-ExecutionProvider.md]: not sure how much this should be explicitly mentioned since it is a higher-level component

the functionalities of ORT-EP in isolation are fairly limited when it comes to this feature

generally had this debate within myself for mentioning higher-level components related to this feature - e.g. Olive, olive-recipes, onnxruntime-genai - given the above

File: docs/execution_providers/QNN-ExecutionProvider.md [hunk 2]
```diff
@@ -82,6 +85,15 @@ The QNN Execution Provider supports a number of configuration options. These pro
 **Note:** `backend_path` is an alternative to `backend_type`. At most one of the two should be specified.
 `backend_path` requires a platform-specific path (e.g., `libQnnCpu.so` vs. `QnnCpu.dll`) but also allows one to specify an arbitrary path.
 
+|`"genie_log_level"`|Description|
```
Comment (qti-kromero) [inline docs/execution_providers/QNN-ExecutionProvider.md:88]: debated putting this in a separate GenAI section or similar. also thought about doing the same under the dynamic option section. however, not sure if that fits with the whole "Genie as another backend" model. open to thoughts

File: docs/execution_providers/QNN-ExecutionProvider.md [hunk 3]
```diff
@@ -580,6 +593,81 @@ del session
...
+
+# Load the ONNX model that wraps the Genie DLC
+sess = ort.InferenceSession("model.onnx", sess_options=sess_options)
+
+# Reset the KV-cache at the start of a new conversation
+run_options = ort.RunOptions()
+run_options.add_run_config_entry("kvcache_rewind", "0")
+
+# Run one auto-regressive step: input = token IDs, output = logits
+input_tokens = np.array([[42, 73, 101]], dtype=np.int32)
```
Comment (qti-kromero) [inline docs/execution_providers/QNN-ExecutionProvider.md]: need to look at dimensions for actual models that I have currently

File: docs/execution_providers/QNN-ExecutionProvider.md [hunk 4]
```diff
@@ -580,6 +593,81 @@ del session
...
+logits = sess.run(None, {"input_ids": input_tokens}, run_options)[0]
+
+# Subsequent steps reuse the KV-cache (no rewind needed)
+next_tokens = np.array([[202]], dtype=np.int32)
```
Comment (qti-kromero) [inline docs/execution_providers/QNN-ExecutionProvider.md]: same here

File: docs/execution_providers/QNN-ExecutionProvider.md (hunk 2 above)
Comment (tirupath-qti) [reply docs/execution_providers/QNN-ExecutionProvider.md:88]: do we need separate genie_log_level ? can't we leverage ort log level flag?

File: docs/execution_providers/QNN-ExecutionProvider.md (hunk 1 above)
Comment (qti-kromero) [reply docs/execution_providers/QNN-ExecutionProvider.md]: For complete chat functionality, including tokenization, sampling, autoregressive looping, etc., a higher-level framework such as onnxruntime-genai should be used.

Comment (qti-kromero) [reply docs/execution_providers/QNN-ExecutionProvider.md]: also check onnxruntime-genai docs to see if changes are needed there to describe this feature

**Review by tirupath-qti — APPROVED** [review]

File: docs/execution_providers/QNN-ExecutionProvider.md (hunk 1 above)
Comment (qti-kromero) [reply docs/execution_providers/QNN-ExecutionProvider.md]: created https://jira-dc.qualcomm.com/jira/browse/AISW-181801 to update onnxruntime-genai docs

File: docs/execution_providers/QNN-ExecutionProvider.md (hunk 2 above)
Comment (qti-kromero) [reply docs/execution_providers/QNN-ExecutionProvider.md:88]: creating the following ticket to follow-up on this: https://jira-dc.qualcomm.com/jira/browse/AISW-181802
